### ISSUE TITLE
[explicit-module-boundary-types] allowedNames doesn't work for arrow functions

### ISSUE THREAD

**Issue by NicBright** [typescript-eslint/typescript-eslint#4463]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

See the following screenshot that directly illustrates the problem:
![image](https://user-images.githubusercontent.com/1225659/150336157-8ea18b6a-9412-4926-aec1-0bd2e9a34e30.png)

```typescript
export class Repro {
    selectState = () => {
        someStateProp: 5;
    };
}
```

.eslintrc.json extract:
```json
"@typescript-eslint/explicit-module-boundary-types": ["error", {
                    "allowedNames": ["selectState"]
                }]
```

**Expected Result**
The rule should allow my public selectState instance property to be assigned an untyped arrow function.

**Actual Result**

The rule disallows the assignment (see screenshot above).

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.10.0` |
| `@typescript-eslint/parser`        | `5.10.0` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.17.0` |
